core/controllers/subtopic_viewer.py

```diff
@@ -59,17 +59,22 @@ def get(self, topic_name, subtopic_id):
 
         subtopic_id = int(subtopic_id)
         topic = topic_fetchers.get_topic_by_name(topic_name)
-        for subtopic in topic.subtopics:
+        next_subtopic = None
+        for index, subtopic in enumerate(topic.subtopics):
             if subtopic.id == subtopic_id:
                 subtopic_title = subtopic.title
+                if index != len(topic.subtopics) - 1:
+                    next_subtopic = topic.subtopics[index + 1].to_dict()
```

Comment:
Naming is wrong -- this suggests a domain object. Probably you'll want it to be next_subtopic_dict.

Do you need to pass the whole subtopic, though? If possible, it would be better to pass just the minimal subset of data required.

Reply:
I need to construct a subtopic object with the dictionary, so would probably need all of it

Reply:
Done.